eureka2-write-server/src/main/java/com/netflix/eureka2/server/service/bootstrap/PeerRegistryBootstrapService.java

```diff
@@ -0,0 +1,118 @@
+import rx.schedulers.Schedulers;
+
+/**
+ * @author Tomasz Bak
+ */
+@Singleton
+public class PeerRegistryBootstrapService implements RegistryBootstrapService {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeerRegistryBootstrapService.class);
+
+    private final EurekaEndpointResolver bootstrapResolver;
+    private final Scheduler scheduler;
+
+    @Inject
+    public PeerRegistryBootstrapService(PeerBootstrapResolverProvider bootstrapResolverProvider) {
+        this(bootstrapResolverProvider.get(), Schedulers.computation());
+    }
+
+    public PeerRegistryBootstrapService(EurekaEndpointResolver bootstrapResolver, Scheduler scheduler) {
+        this.bootstrapResolver = bootstrapResolver;
+        this.scheduler = scheduler;
+    }
+
+    @Override
+    public Observable<Void> loadIntoRegistry(final SourcedEurekaRegistry<InstanceInfo> registry, final Source source) {
+        return bootstrapResolver.eurekaEndpoints()
+                .compose(ChangeNotifications.<EurekaEndpoint>buffers())
+                .compose(ChangeNotifications.<EurekaEndpoint>snapshots())
+                .take(1)
+                .flatMap(new Func1<LinkedHashSet<EurekaEndpoint>, Observable<EurekaEndpoint>>() {
+                    @Override
+                    public Observable<EurekaEndpoint> call(LinkedHashSet<EurekaEndpoint> eurekaEndpoints) {
+                        return Observable.from(eurekaEndpoints);
+                    }
+                }).flatMap(new Func1<EurekaEndpoint, Observable<Boolean>>() {
+                    @Override
+                    public Observable<Boolean> call(EurekaEndpoint endpoint) {
+                        return loadRegistryFrom(new Server(endpoint.getHostname(), endpoint.getInterestPort()), registry, source);
+                    }
+                })
+                .filter(new Func1<Boolean, Boolean>() { // true - download succeeded, false - failed, so try another server
+                    @Override
+                    public Boolean call(Boolean status) {
+                        return status;
+                    }
+                }).defaultIfEmpty(false).take(1).flatMap(new Func1<Boolean, Observable<Void>>() {
+                    @Override
+                    public Observable<Void> call(Boolean status) {
+                        if (status) {
+                            logger.info("Bootstrap from peer completed successfully");
+                            return Observable.empty();
+                        }
+                        logger.error("Cannot bootstrap registry from peers");
+                        return Observable.error(new Exception("Cannot bootstrap registry from peers"));
+                    }
+                });
+    }
+
+    private Observable<Boolean> loadRegistryFrom(final Server server, final SourcedEurekaRegistry<InstanceInfo> registry, final Source source) {
+        return createLightEurekaInterestClient(server).forInterest(Interests.forFullRegistry())
```

Comment:
This here seems to be doing almost exactly what the FullFetchInterestClient does for the read servers. Can we not reuse the FullFetchInterestClient and pass it the write server's registry?